Hypernova/crossplatform

chasester opened the pull request:
Basic approval

api-Hypernova committed:
fixes for GCC

api-Hypernova committed:
fixes for GCC

api-Hypernova committed:
Enable exceptions

api-Hypernova committed:
Enable exceptions

api-Hypernova committed:
Random updates

api-Hypernova committed:
Random updates

api-Hypernova committed:
some fixes

api-Hypernova committed:
some fixes

api-Hypernova committed:
A bunch of updates

api-Hypernova committed:
A bunch of updates

api-Hypernova committed:
Fix const passing

api-Hypernova committed:
Fix const passing

api-Hypernova committed:
Update makefile

api-Hypernova committed:
Update makefile

api-Hypernova committed:
Makefile

api-Hypernova committed:
Makefile

api-Hypernova committed:
-framework

api-Hypernova committed:
-framework

api-Hypernova committed:
Random updates

api-Hypernova committed:
Random updates

api-Hypernova committed:
Makefile and serializer updates for clang

api-Hypernova committed:
Makefile and serializer updates for clang

api-Hypernova committed:
Progress updates

api-Hypernova committed:
Progress updates

api-Hypernova committed:
Removed nonexistant object files from Makefile

api-Hypernova committed:
Removed nonexistant object files from Makefile

chasester committed:
Fixed errors in the make file and moved nodes.h/cpp to shared

chasester committed:
Fixed errors in the make file and moved nodes.h/cpp to shared

api-Hypernova commented:
Let's not merge this yet until I can get something stable. It still uses some hacks which I don't think are suitable for master yet.

chasester commented:
is that what that thingy does 0.0 
i was just messing around with shit again :+1:

chasester committed:
fixed ignore list to add a few extra things

(exe bak lib map)

api-Hypernova committed:
Updates

api-Hypernova committed:
Updates

api-Hypernova committed:
Merge fix

chasester committed:
Fixed file paths and added organization, fixed make file agian

Added config to the main directory so program will run when built. But
all *.halo files into a directory halo both in vcpp and in the main
directory. Add some code to allow for loading in this new directory.

chasester committed:
Fixed file paths and added organization, fixed make file agian

Added config to the main directory so program will run when built. But
all *.halo files into a directory halo both in vcpp and in the main
directory. Add some code to allow for loading in this new directory.

api-Hypernova committed:
Removed engine/wings.cpp from Makefile

api-Hypernova committed:
Removed engine/wings.cpp from Makefile

api-Hypernova committed:
yay, final updates

api-Hypernova committed:
yay, final updates

chasester committed:
added a few vs dependent changes

needed to change it so shared\node.h,cpp exist instead of
game\node.h,cpp

chasester committed:
added a few vs dependent changes

needed to change it so shared\node.h,cpp exist instead of
game\node.h,cpp

api-Hypernova reviewed (commented):
Seems fine, go ahead.

api-Hypernova committed:
Small updates to makefile, tools.h and wings.cpp

api-Hypernova committed:
Small updates to makefile, tools.h and wings.cpp

chasester commented on source/shared/tools.h:
```diff
@@ -20,11 +20,11 @@ typedef unsigned long long int ullong;
-#else
+//#ifdef _DEBUG
+//#define ASSERT(c) assert(c)
+//#else
```
we need to change this back

chasester commented on source/shared/wings.cpp:
```diff
@@ -406,18 +406,18 @@ void config(asIScriptEngine *asEngine){
     //r = asEngine->RegisterObjectMethod("node", "void seteditbox(int i)", asMETHOD(node, seteditbox), asCALL_THISCALL); assert(r >= 0);
 	//rigid body and physics
-    r = asEngine->RegisterObjectType("rigidbody", sizeof(rigidbody), asOBJ_VALUE | asOBJ_APP_CLASS_CDAK); assert(r >= 0);
+    //r = asEngine->RegisterObjectType("rigidbody", sizeof(rigidbody), asOBJ_VALUE | asOBJ_APP_CLASS_CDAK); assert(r >= 0);
```
delete this line we wont need it

chasester commented on source/shared/wings.cpp:
```diff
@@ -406,18 +406,18 @@ void config(asIScriptEngine *asEngine){
     //r = asEngine->RegisterObjectBehaviour("node", asBEHAVE_GET_WEAKREF_FLAG, "int &f()", asMETHOD(node, getweakflagref), asCALL_THISCALL); assert(r >= 0);
 
-	r = asEngine->RegisterObjectProperty("node", "string name", asOFFSET(node, name));
+    //r = asEngine->RegisterObjectProperty("node", "string name", asOFFSET(node, name));
```
will need to add back when we fix the string issue or nodes wont init properly

chasester commented on source/shared/wings.cpp:
```diff
@@ -406,18 +406,18 @@ void config(asIScriptEngine *asEngine){
     r = asEngine->RegisterObjectMethod("rigidbody", "void aabbfrommodel(int index)", asMETHOD(rigidbody, rigidbody::aabbfrommodel), asCALL_THISCALL); assert(r >= 0);
     r = asEngine->RegisterGlobalFunction("void addrigidbody(const rigidbody &in, const node @+ to)", asFUNCTIONPR(addrigidbody, (rigidbody&, node*), void), asCALL_CDECL); asFATAL(r);
-	r = asEngine->RegisterGlobalFunction("void createobject(string s, const vec &in, const vec &in)", asFUNCTIONPR(createobject, (str, vec&, vec&), void), asCALL_CDECL); asFATAL(r);
+    //r = asEngine->RegisterGlobalFunction("void createobject(string s, const vec &in, const vec &in)", asFUNCTIONPR(createobject, (str, vec&, vec&), void), asCALL_CDECL); asFATAL(r);
```
will need to add back when we fix the string issue or nodes wont init properly

chasester commented:
we need to come up with one gitignore list.

chasester commented on source/Makefile:
```diff
@@ -71,7 +71,6 @@ CLIENT_OBJS= \
 	shared/scriptbuilder.o\
 	shared/scripthandle.o\
 	shared/scripthelper.o\
-	shared/scriptstring.o\
```
doesn exist it is just miss spelled, should be scriptstringd.o

api-Hypernova committed:
Remove -framework and more Mac specific dependency handling. Mac users should now install all SDL dependencies through brew

api-Hypernova committed:
Remove -framework and more Mac specific dependency handling. Mac users should now install all SDL dependencies through brew

api-Hypernova committed:
Added shared/halo.h

api-Hypernova committed:
Added shared/halo.h

api-Hypernova committed:
Too many updates to list

api-Hypernova committed:
Local commit

mufeedali commented:
nightglare here, are you done with the clean-ups yet?

api-Hypernova committed:
Shader crash update

api-Hypernova committed:
Branch merge with shader crash update

api-Hypernova committed:
Removed client/server executables

api-Hypernova committed:
WIP build scripts

chasester commented:
Depricated commits, we can use this for reference, in the future for additiona cross platform support and changes, but all othese changes would not reflect properly onto the current version of the project.
